src/module_3/module_3.py

```diff
@@ -0,0 +1,129 @@
+# %% Import packages
+import pandas as pd
+
+from sklearn.model_selection import train_test_split
+from sklearn import linear_model
+from sklearn.linear_model import Ridge
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.ensemble import RandomForestClassifier
+from sklearn import metrics
+from sklearn.preprocessing import normalize
+
+import matplotlib.pyplot as plt
+
+# %% Load data
+address = "D:\\Yo\\"
+filename = "feature_frame.csv"
+df = pd.read_csv(address + filename)
+
+# %% Data preprocessing
+
+# Obtain the id's for the orders where 5 or more products where bought
+filter_df = df[df["outcome"]==1]
+filter_df = filter_df.groupby("order_id").filter(lambda x: len(x) > 4)
+
+# Filter the data so we retain only those ids
+dataset = df[df["order_id"].isin(filter_df["order_id"])]
+
+# Change date format to retain only year, month and day
+dataset["created_at"] = dataset["created_at"].str[:-9]
+dataset["created_at"] = dataset["created_at"].str.replace("-","")
+pd.to_numeric(dataset["created_at"])
+dataset["order_date"] = dataset["order_date"].str[:-9]
+dataset["order_date"] = dataset["order_date"].str.replace("-","")
+pd.to_numeric(dataset["order_date"])
+
+# Order by date
+dataset = dataset.sort_values(by=["order_date"])
+
+# %% Drop qualitative features
+print(dataset.columns)
+print(dataset.dtypes)
+dataset.drop(["product_type" , "vendor"], axis=1, inplace=True)
+
+# %% Divide the data in training, validation and test set
+def timesplit_by_order_id(data: pd.DataFrame, train_size: float, validation_test_size: float):
+    total_orders = data.order_id.unique()
+    train_num = round(total_orders.size*train_size)
+    val_num = round(total_orders.size*validation_test_size)
+    test_num = val_num
+    gap = round(total_orders.size*(1-validation_test_size-train_size)/3)
+
+    train_orders=total_orders[0:train_num]
+    val_orders=total_orders[train_num+gap:train_num+gap+val_num]
+    test_orders=total_orders[train_num+2*gap+val_num:train_num+2*gap+val_num+test_num]
```

Comment:
this looks very complicated, I would do (supposing that I'm understanding correctly what are you trying to do) something like:
```
train_orders=total_orders[:train_num]
val_orders =total_orders[train_num:train_num+val_num]
test_orders =total_orders[train_num+val_num:]
```

Reply:
Also, very very important, if you are doing a random train / val / test split ALWAYS ensure that your dataset has been shuffled before (something like `total_orders = total_orders.sample(frac = 1)`) otherwise train / val / test may not have the same distribution